StorageAccountManagement/Program.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Azure;
+using Azure.Core;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Storage;
+using Azure.ResourceManager.Storage.Models;
+
+using StorageAccountManagement;
+
+// Update placeholders with your own values - store values securely
+const string subscriptionId = "<subscription-id>";
+const string rgName = "<resource-group-name>";
+string storageAccountName = "<storage-account-name>";
+AzureLocation location = AzureLocation.EastUS;
+
+// Provide configuration options for the ArmClient
+ArmClientOptions armClientOptions = new()
+{
+    Retry = {
+        Delay = TimeSpan.FromSeconds(2),
+        MaxRetries = 5,
+        Mode = RetryMode.Exponential,
+        MaxDelay = TimeSpan.FromSeconds(10),
+        NetworkTimeout = TimeSpan.FromSeconds(100)
+    },
+};
+
+// Authenticate to Azure and create the top-level ArmClient
+ArmClient armClient = new(new DefaultAzureCredential(), subscriptionId, armClientOptions);
+
+// Create a resource identifier and get the subscription resource
+ResourceIdentifier resourceIdentifier = new($"/subscriptions/{subscriptionId}");
+SubscriptionResource subscription = armClient.GetSubscriptionResource(resourceIdentifier);
+
+// Register the Storage resource provider in the subscription
+await ManagementTasks.RegisterSRPInSubscription(subscription);
+
+// Create a new resource group - if one already exists then it's updated
+ArmOperation<ResourceGroupResource> rgOperation = await subscription
+    .GetResourceGroups()
+    .CreateOrUpdateAsync(WaitUntil.Completed, rgName, new ResourceGroupData(location));
+ResourceGroupResource resourceGroup = rgOperation.Value;
+
+// Check if the account name is available
+bool? nameAvailable = subscription
+    .CheckStorageAccountNameAvailability(new StorageAccountNameAvailabilityContent(storageAccountName)).Value.IsNameAvailable;
+
+StorageAccountResource storageAccount = 
+    await ManagementTasks.CreateStorageAccount(resourceGroup, storageAccountName);
+
+// Get all the storage accounts for a given subscription
+await ManagementTasks.ListStorageAccountsForSubscription(subscription);
+
+// Get a list of storage accounts within a specific resource group
+await ManagementTasks.ListStorageAccountsInResourceGroup(subscription.GetResourceGroupAsync(rgName).Result);
```

Comment:
For the get/list out accounts, we might can add some code to show how to utilize the result.

Reply:
The method implementation shows how to iterate the results and prints them to the console, as an example. Did you have something else in mind?

Reply:
I see. The list result is utilized in the function.
Thanks for the explanation!